src/count-islands/index.js

```diff
@@ -1,3 +1,47 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  let count = 0
+  let list = {}
+
+  grid.forEach((element, rootI) => {
+    element.forEach((item, childI) => {
+      list[rootI.toString() + childI.toString()] = item
+    })
+  });
+
+  grid.forEach((element, rootIndex) => {
+    element.forEach((item, childIndex) => {
+      if (item == 1 &&
+        list[rootIndex.toString() + childIndex.toString()] !== null) {
+        count++
+        crawler(rootIndex,childIndex)
+      }
+    })
+  });
+
+  function crawler(row, col) {
```

Comment:
move this crawler function out of the countIsLand() function

Reply:
done